x-pack/plugins/monitoring/public/application/pages/logstash/pipelines.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useContext, useState, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { find } from 'lodash';
+import { useKibana, useUiSetting } from '../../../../../../../src/plugins/kibana_react/public';
+import { GlobalStateContext } from '../../global_state_context';
+import { ComponentProps } from '../../route_init';
+import { useCharts } from '../../hooks/use_charts';
+// @ts-ignore
+import { isPipelineMonitoringSupportedInVersion } from '../../../lib/logstash/pipelines';
+// @ts-ignore
+import { PipelineListing } from '../../../components/logstash/pipeline_listing/pipeline_listing';
+import { LogstashTemplate } from './logstash_template';
+import { useTable } from '../../hooks/use_table';
+
+export const LogStashPipelinesPage: React.FC<ComponentProps> = ({ clusters }) => {
+  const globalState = useContext(GlobalStateContext);
+  const { onBrush } = useCharts();
+  const { services } = useKibana<{ data: any }>();
+  const dateFormat = useUiSetting<string>('dateFormat');
+  
+  const clusterUuid = globalState.cluster_uuid;
+  const ccs = globalState.ccs;
+
+  const cluster = find(clusters, {
+    cluster_uuid: clusterUuid,
+  });
+  const [data, setData] = useState(null);
+  const { getPaginationTableProps, getPaginationRouteOptions, updateTotalItemCount } =
+    useTable('logstash.pipelines');
+
+  const title = i18n.translate('xpack.monitoring.logstash.overview.title', {
+    defaultMessage: 'Logstash',
+  });
+
+  const pageTitle = i18n.translate('xpack.monitoring.logstash.overview.pageTitle', {
+    defaultMessage: 'Logstash overview',
+  });
+
+  const getPageData = useCallback(async () => {
+    const bounds = services.data?.query.timefilter.timefilter.getBounds();
+    const url = `../api/monitoring/v1/clusters/${globalState.cluster_uuid}/logstash/pipelines`;
+    
+    const options: any = getPaginationRouteOptions();
+    const response = await services.http?.fetch(url, {
+      method: 'POST',
+      body: JSON.stringify({
+        ccs,
+        timeRange: {
+          min: bounds.min.toISOString(),
+          max: bounds.max.toISOString(),
+        },
+        pagination: options.pagination,
+        queryText: options.queryText,
+      }),
+    });
+
+    setData(response);
+    updateTotalItemCount(response.totalPipelineCount);
+  }, [ccs, clusterUuid, services.data?.query.timefilter.timefilter, services.http, getPaginationRouteOptions, updateTotalItemCount]);
+
+  const renderOverview = (pageData: any) => {
+    if (pageData === null) {
+      return null;
+    }
+    const { clusterStatus, pipelines } = pageData || {};
```

Comment:
nit: the fallback to an empty object breaks line 74 when reading `clusterStatus.versions`. Can we make the initial `pageData` check more robust (or maybe null is enough) and get rid of fallbacks and conditionals in the following lines ?